grpc.go

```diff
@@ -578,6 +607,24 @@ func (a *agentGRPC) CreateContainer(ctx context.Context, req *pb.CreateContainer
 		return emptyResp, err
 	}
 
+	if a.sandbox.guestHookPath != "" {
+		// Add any custom OCI hooks to the spec
+		pb.AddGuestHooks(ociSpec, a.sandbox.guestHookPath)
```

Comment:
Now I see you have pushed the hookpath to runtime configuration, still please only scan the hook path **one time only** and save the result in the sandbox data structure. You can do it in the `CreateSandbox` handler instead of `CreateContainer` here.

Reply:
Agree. Please move the logic to scan the hooks to CreateSandbox.

Reply:
Ok now it should only do the directory lookup once.